src/main.rs

```diff
@@ -0,0 +1,270 @@
+extern crate clap;
+extern crate crypto;
+#[macro_use]
+extern crate nickel;
+extern crate nickel_mustache;
+extern crate rustc_serialize;
+extern crate rand;
+extern crate postgres;
+
+
+use std::path::Path;
+use std::ascii::AsciiExt;
+use std::collections::HashMap;
+use std::error::Error;
+use std::fs::File;
+use std::fmt;
+use std::io::prelude::*;
+use rand::os::OsRng;
+use rand::Rng;
+
+use nickel::status::StatusCode;
+use nickel::{Nickel, HttpRouter, FormBody};
+use nickel_mustache::Render;
+
+use clap::{App, Arg};
+
+use crypto::bcrypt;
+
+use postgres::{Connection, TlsMode};
+
+
+fn main() {
+    let matches = App::new("synapse-password-reset")
+        .version("v0.0.1")
+        .author("Euan Kemp <euank@euank.com>")
+        .args(&[Arg::with_name("token-dir")
+                    .help("sets the database directory to use")
+                    .takes_value(true)
+                    .short("t")
+                    .long("token-dir")
+                    .required(true),
+                Arg::with_name("pepper")
+                    .help("sets the hash pepper (e.g. from your synapse config)")
+                    .takes_value(true)
+                    .short("p")
+                    .long("pepper")
+                    .required(true),
+                Arg::with_name("db")
+                    .help("sets the postgres db to connect to (including username,pass)")
```

Comment:
Is this a database URI? It would make it more clear if it showed an example format here.